carl/envs/rna/carl_rna.py

```diff
@@ -0,0 +1,119 @@
+from carl.envs.carl_env import CARLEnv
+from carl.envs.rna.parse_dot_brackets import parse_dot_brackets
+from carl.envs.rna.rna_environment import (
+    RnaDesignEnvironment,
+    RnaDesignEnvironmentConfig,
+)
+import numpy as np
+from typing import Optional, Dict, Union, List
+from carl.utils.trial_logger import TrialLogger
+
+from carl.envs.rna.carl_rna_definitions import (
+    DEFAULT_CONTEXT,
+    ACTION_SPACE,
+    OBSERVATION_SPACE,
+    CONTEXT_BOUNDS,
+)
+
+from carl.utils.trial_logger import TrialLogger
+from carl.utils.types import Context, Contexts
+from carl.context.selection import AbstractSelector
+
+import gym
+
+
+class CARLRnaDesignEnv(CARLEnv):
+    def __init__(
+        self,
+        env: gym.Env = None,
+        data_location: str = "carl/envs/rna/learna/data",
+        contexts: Contexts = {},
+        hide_context: bool = False,
+        add_gaussian_noise_to_context: bool = False,
+        gaussian_noise_std_percentage: float = 0.01,
+        logger: Optional[TrialLogger] = None,
+        scale_context_features: str = "no",
+        default_context: Optional[Context] = DEFAULT_CONTEXT,
+        max_episode_length: int = 500,
+        state_context_features: Optional[List[str]] = None,
+        context_mask: Optional[List[str]] = None,
+        dict_observation_space: bool = False,
+        context_selector: Optional[
+            Union[AbstractSelector, type[AbstractSelector]]
+        ] = None,
+        context_selector_kwargs: Optional[Dict] = None,
+    ):
+        """
+        Parameters
+        ----------
+        env: gym.Env, optional
+            Defaults to classic control environment mountain car from gym (MountainCarEnv).
+        contexts: List[Dict], optional
+            Different contexts / different environment parameter settings.
+        instance_mode: str, optional
+        """
+        if not contexts:
+            contexts = {0: DEFAULT_CONTEXT}
+        if env is None:
+            env_config = RnaDesignEnvironmentConfig(
+                mutation_threshold=DEFAULT_CONTEXT["mutation_threshold"],
+                reward_exponent=DEFAULT_CONTEXT["reward_exponent"],
+                state_radius=DEFAULT_CONTEXT["state_radius"],
+            )
+            dot_brackets = parse_dot_brackets(
+                dataset=DEFAULT_CONTEXT["dataset"],
+                data_dir=data_location,
+                target_structure_ids=DEFAULT_CONTEXT["target_structure_ids"],
+            )
+            env = RnaDesignEnvironment(dot_brackets, env_config)
+
+        env.action_space = ACTION_SPACE
```

Comment:
TODO: pass action and observation space via argument

Reply:
```python
def __init__(self, ..., action_space: gym.spaces.Discrete = ACTION_SPACE, ....)
```

Reply:
Also shouldn't this be passed to `RnaDesignEnvironment`?

Reply:
```RnaDesignEnvironment``` does not take action space as an input, but it is set right after the environment is initialized with the config. So, at the moment the wrapper is doing a lot of heavy lifting, and it is partly because I didn't want to touch the environment itself too much. 

We could of course standardize the interface a bit more, but my question would then be: Given Frederic is working on a new version, does it make sense to do it rn?